hydromt_delft3dfm/workflows/boundaries.py

```diff
@@ -638,3 +625,280 @@ def compute_meteo_forcings(
     da_out.dropna(dim="time")
 
     return da_out
+
+
+def _standardize_forcing_timeindexes(da):
+    """Standardize timeindexes frequency based on forcing DataArray"""
+    _TIMESTR = {"D": "days", "H": "hours", "T": "minutes", "S": "seconds"}
+    dt = pd.to_timedelta((da.time[1].values - da.time[0].values))
+    freq = dt.resolution_string
+    multiplier = 1
+    if freq == "D":
+        logger.warning(
+            "time unit days is not supported by the current GUI version: 2022.04"
+        )  # converting to hours as temporary solution # FIXME: day is converted to hours temporarily
+        multiplier = 24
+    if len(pd.date_range(da.time[0].values, da.time[-1].values, freq=dt)) != len(
+        da.time
+    ):
+        logger.error("does not support non-equidistant time-series.")
+    freq_name = _TIMESTR[freq]
+    freq_step = getattr(dt.components, freq_name)
+    bd_times = np.array([float(i * freq_step) for i in range(len(da.time))])
+    if multiplier == 24:
+        bd_times = np.array([(i * freq_step * multiplier) for i in range(len(da.time))])
+        freq_name = "hours"
+    return bd_times, freq_name
+
+
+def compute_forcing_values_points(
+    gdf: gpd.GeoDataFrame,
+    da: xr.DataArray = None,
+    forcing_value: float = 0.0,
+    forcing_type: str = "lateral_discharge",
+    forcing_unit: str = "m3/s",
+    logger=logger,
+):
+    """
+    Compute 1d forcing values.
+
+    Used for 1D lateral point locations.
+
+    Parameters
+    ----------
+    gdf : gpd.GeoDataFrame
+        GeoDataFrame of points to add 1D forcing data.
+
+        * Required variables: ['geometry']
+    da : xr.DataArray, optional
+        xr.DataArray containing the forcing timeseries values.
+        If None, uses a constant ``forcing_value`` for all forcings.
+
+        * Required variables: ['forcing_type']
+
+    forcing_value : float, optional
+        Constant value to use for all forcings if ``da`` is None and to
+        fill in missing data.
+        By default 0.0 ``forcing_unit``
+    forcing_type : {'lateral_discharge'}
+        Type of forcing to use.
+        For now only support 'lateral_discharge'.
+        By default 'lateral_discharge'
+    forcing_unit : {'m3/s'}
+        Unit corresponding to ``forcing_type``.
+        By default 'm3/s'
+    logger
+        Logger to log messages.
+    """
+    # TODO: harmonize for other point forcing #21
+    # first process data based on either timeseries or constant
+    # then update data based on either nodes or branches
+    # Timeseries forcing values
+    if da is not None:
+        logger.info(f"Preparing 1D forcing type {forcing_type} from timeseries.")
+
+        # get forcing data freq in seconds
+        bd_times, freq_name = _standardize_forcing_timeindexes(da)
+
+        # instantiate xr.DataArray for forcing data
+        # NOTE only support points on branches
+        da_out = xr.DataArray(
```

Comment:
Can you use this instead? da_out = hydromt.vector.GeoDataset.from_gdf(gdf, da)

Reply:
And then still need to set attrs and add new coords for branchid and chainage?

Reply:
After trying hydromt.vector.GeoDataArray.from_gdf, I noticed that we still need to update time, add x and y, and set attributes to the geodataarray. At current stage I am not sure there is too much to gain to replace. 
However, netcdf file is being supported by Delft3D FM. then we dont have to explictly compute the time indexes and many of the attributes will not be needed anymroe. 
I propose we leave it as todo when delft3d FM supports netcdf files (in testing phase)